pyproject.toml

```diff
@@ -20,12 +20,12 @@ dependencies = [
     "sympy >= 1.2",
     "termcolor >= 1.0.0",
     "textx >= 2.2",
-    "astunparse >= 1.6.0", # astunparse is only needed for Python3.8, we should use ast.unparse when we drop Python3.8.
     "packaging",
 ]
 
 [project.optional-dependencies]
 test = [
+    "astunparse >= 1.6.0",
```

Comment:
Why do we need `astunparse` for the tests?

Reply:
It is used to test the comments here:
https://github.com/pyccel/pyccel/blob/b3ad4a99e6e4e6a05e0ba9c76bbcdabb5ead11f1/tests/parser/test_comments.py#L9-L14

Reply:
I see. But maybe there is a similar class in the `ast` package?

Reply:
I didn't see anything in the `ast` documentation

Reply:
There is nothing about `Unparser` in the `astunparse` documentation either 😅 ... We are using library internals here...

Reply:
Here is the source code:
- https://github.com/simonpercivall/astunparse/blob/2acce01fcdda2ea32eea835c30ccca21aaff7297/lib/astunparse/unparser.py#L27
- https://github.com/python/cpython/blob/b93b7e566e5a4efe7f077af2083140e50bd2b08f/Lib/ast.py#L666

Reply:
I have just done a quick test. It seems this is possible but it is a lot more fiddly to do this test cleanly with `ast.unparse` instead of `astunparse` as `ast.unparse` is lower-level so it preserves less syntactic information. E.g I am currently getting the error:
```
AssertionError: assert 'b=(a+2)' == 'b=a+2'
```
If we are only using it to test comments though then maybe we don't care and I just need to remove unnecessary parentheses from the test file?

Reply:
They seem fairly unnecessary so maybe they were only added to match what astunparse expected? Either that or it was to ensure readability when ` ` are stripped out

Reply:
Similarly `ast.unparse` parses this:
https://github.com/pyccel/pyccel/blob/b3ad4a99e6e4e6a05e0ba9c76bbcdabb5ead11f1/tests/parser/scripts/comments.py#L60-L61
as:
```python
elif d + 1:
```

Reply:
Indeed there is no guarantee that the original source code is preserved, so we cannot make a string comparison. 

Maybe we can just check the commented lines, as this is what the unit test is about...

Reply:
Fixed: f0ccfd5d2

But I am not sure that this test is the most robust (but it wasn't before either)

Reply:
> Maybe we can just check the commented lines, as this is what the unit test is about...

The problem is that they are not so easy to identify after some of these simplifications as the line where they are found can change

Reply:
I think your solution is good enough for now!